packages/coreutils/src/pageconfig.ts

```diff
@@ -110,17 +110,18 @@ export namespace PageConfig {
   }
 
   /**
-   * Get the base url for a Jupyter application.
+   * Get the base url for a Jupyter application, or an empty string.
+   *
+   * #### Notes
+   * If the baseUrl is not provided in `jupyter-config-data`, `location.origin`
+   * is used (if in a browser environment).
    */
   export function getBaseUrl(): string {
-    let baseUrl = getOption('baseUrl');
-    if (!baseUrl || baseUrl === '/') {
-      baseUrl =
-        typeof location === 'undefined'
-          ? 'http://localhost:8888/'
-          : location.origin + '/';
+    const baseUrl = URLExt.normalize(getOption('baseUrl'));
+    if (!baseUrl && typeof location !== 'undefined') {
+      return URLExt.normalize(location.origin);
```

Comment:
What happens if you simply use `URLExt.normalize('/')` here, and skip the location check? Will this work on node as well?

Reply:
It gives back `/` in node, which would go to the root of your server.  However, that doesn't help us get a `wsUrl`...